src/playground/sample_data_tooling/sample_data_tests/sample_data_generator_tests/unit-test/sample_data_generator_unit_test.py

```diff
@@ -0,0 +1,243 @@
+import sys
+import json
+import pytest
+from datetime import date
+from types import NoneType
+
+# Adds parent directory sample_data_tooling to sys.path
+sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))))
+import sample_data_generator.sample_data_generator as generator
+
+
+# This path is for getting the parent directory sample_data_generator_tests
+FILE_PATH = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+
+# Sample inputs (valid)
+valid_json_shorthand = {
+        "year": "year",
+        "random number": "integer",
+        "email": "email",
+        }
+
+valid_json_mapping = {"properties": {
+        "first name":    { "type" : ["first_name"]},
+        "last name":     { "type" : "last_name"},
+        "companies":{ "type" : ["array", "company", 5] }
+        }}
+
+valid_json_string_shorthand = json.dumps({
+        "year": "year",
+        "addresses": ["array", "address", "integer", [1, 3]],
+        "director": "name"
+        })
+
+valid_json_string_mapping = json.dumps({"properties": {
+        "year":    { "type" : ["year"]},
+        "addresses":     { "type" : ["array", "address", "integer", (1, 3)] },
+        "director":{ "type" : "name" }
+        }})
+
+valid_json_string_mapping_two = json.dumps({"properties": {
+        "year":    { "type" : ["year"]},
+        "integers":     { "type" : ["array", "integer", "integer", (1, 3), 5, 10] },
+        "director":{ "type" : "name" }
+        }})
+
+
+# Sample inputs (invalid)
+invalid_json_shorthand = {
+        "year": "year",
+        "random number": "integer",
+        "email": "emaiz",
+        }
+
+invalid_empty_json = {}
+
+invalid_json_shorthand_with_array = {
+        "year": ["year", 5],
+        "addresses": ["array", "address", "integer", [1, 3]],
+        "director": "name"
+        }
+
+invalid_json_field_type = {
+        "year": "y",
+        "addresses": ["array", "address", "5", [1, 3]],
+        "director": ["integer", 6, 5]
+        }
+
+
+# Testing each function of generator.py
+def test_valid_choose_field():
+        # Tests for correct type returned with minimum required arguments
+        assert generator.choose_field("null") == None
+        assert type(generator.choose_field("float")) is float
+        assert type(generator.choose_field("double")) is float
+        assert type(generator.choose_field("integer")) is int
+        assert type(generator.choose_field("keyword", ["email"])) is str
+        assert "@" in generator.choose_field("keyword", ["email"])
+        assert type(generator.choose_field("date detection string")) is bool
+        assert type(generator.choose_field("numeric detection string")) is bool 
+        # One of Faker's standard providers
+        ssn = generator.choose_field("ssn")
+        assert type(ssn) is str and "-" == ssn[3] and "-" == ssn[6]
+
+        # 4 array tests:
+        #       - test_array_: A fixed length array with default field values
```

Comment:
is there any reason that not use "test_array_1" here?

Reply:
Typo, it has been fixed in the latest commit